packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -41,6 +41,41 @@ abstract class Layer extends AbstractNode with DiagnosticableTreeMixin {
   @override
   ContainerLayer get parent => super.parent;
 
+  // Whether this layer has any changes since it was last sent to the engine
+  // using [addToScene]. Initialized to true as a new layer has never called
+  // [addToScene].
+  bool _isDirty = true;
+
+  /// Mark that this layer has changed and we need to upload changes to the
+  /// engine using [addToScene].
+  void markDirty() { _isDirty = true; }
```

Comment:
instead of making this public, we should just have the offset and picture fields be getter/setter pairs and have the setter call this if the new value is different.